### ISSUE TITLE
Signaling and registering for push should run in parallel

### ISSUE THREAD

**Issue by ognjenns** [RestComm/restcomm-android-sdk#767]
Registering for push notification is waiting for success connection to signaling. The signaling and push should not depend on each other. We should change this and run them in parallel.

**Comment by atsakiridis** [comment]
Let's also evaluate if the choice to have separate warning callback for push related configurations issues works out well. If not let's do some more brainstorming on possible improvements

**Comment by atsakiridis** [comment]
@oggie let's not mix this issue with callbacks logic. I opened a separate issue for this: #784. I'll fill you in on the details tomorrow
